### PR TITLE
feat(legacy): add available upgrade column

### PR DESCRIPTION
**Pull request by huwshimi**
## Done

- Add the available upgrade column to the controller list.

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Bolla](/HACKING.md#bolla)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)

### QA steps

- Point your UI to Bolla and go to the controllers list.
- You should see an "Available versions" column and an upgrade should show up (if not let me know and I might need to update the backend again).

## Fixes

Fixes: #2491.

### CODE REVIEW

File: legacy/src/app/partials/nodes-list.html [hunk 1]
```diff
@@ -1357,6 +1358,14 @@ <h4 class="page-header__dropdown-title">Add rack controller</h4>
                 </span>
               </div>
             </td>
+            <td class="p-table__cell" aria-label="Available upgrade">
+              <div
+                data-ng-show="controller.versions.update"
+                title="{$ controller.versions.update.version $}"
+              >
+                {$ controller.versions.update.version $}
```
Comment (Caleb-Ellis) [inline legacy/src/app/partials/nodes-list.html:1366]: Do these need to be given the `&&` treatment, or do angular templates fail "gracefully"? Either way I think we should play it safe with angular and have a simple function that checks that an upgrade is available.

File: legacy/src/app/partials/nodes-list.html [hunk 2]
```diff
@@ -1357,6 +1358,14 @@ <h4 class="page-header__dropdown-title">Add rack controller</h4>
                 </span>
               </div>
             </td>
+            <td class="p-table__cell" aria-label="Available upgrade">
+              <div
+                data-ng-show="controller.versions.update"
```
Comment (Caleb-Ellis) [inline legacy/src/app/partials/nodes-list.html:1363]: `data-ng-if` is almost universally the better option. `data-ng-show` still renders the full markup with an added `ng-hide` class that gives it `display: none`.

**Review by Caleb-Ellis — CHANGES_REQUESTED** [review]

File: legacy/src/app/partials/nodes-list.html (hunk 2 above)
Comment (huwshimi) [reply legacy/src/app/partials/nodes-list.html:1363]: Thanks, I'll fix that up.

File: legacy/src/app/partials/nodes-list.html (hunk 1 above)
Comment (huwshimi) [reply legacy/src/app/partials/nodes-list.html:1366]: Good idea.

Comment (huwshimi) [reply legacy/src/app/partials/nodes-list.html]: Done.

File: legacy/src/app/partials/nodes-list.html (hunk 2 above)
Comment (huwshimi) [reply legacy/src/app/partials/nodes-list.html]: Done.

File: legacy/src/app/controllers/tests/test_nodes_list.js [hunk 3]
```diff
@@ -1391,4 +1391,44 @@ describe("NodesListController", function () {
...
+          },
+        },
+      };
+      expect($scope.getUpgradeVersion(controller)).toBe("1.2.3");
+    });
+
+    it("handles a controller without versions", () => {
+      makeController();
+      const controller = {
+        versions: {},
```
Comment (Caleb-Ellis) [inline legacy/src/app/controllers/tests/test_nodes_list.js:1411]: I think is is supposed to be removed

**Review by Caleb-Ellis — APPROVED** [review]
LGTM! Just one comment about a test

File: legacy/src/app/controllers/tests/test_nodes_list.js (hunk 3 above)
Comment (huwshimi) [reply legacy/src/app/controllers/tests/test_nodes_list.js:1411]: Yep! Thanks.

**Pull request merged**